src/object.ts

```diff
@@ -156,7 +156,9 @@ export const pick = <T, TKeys extends keyof T>(
 ): Pick<T, TKeys> => {
   if (!obj) return {} as Pick<T, TKeys>
   return keys.reduce((acc, key) => {
-    return { ...acc, [key]: obj[key] }
+    if(typeof(obj[key]) !== 'undefined')
+      acc[key] = obj[key]
+    return acc;
```

Comment:
I would use `Object.hasOwn` for this check that way we don't end up ignoring properties that exist but have an undefined value.

For example, I think this test will fail with the current implementation
```ts
test('does not ignore undefined values', () => {
      const result = _.pick({ a: 2, b: undefined }, ['b'])
      assert.deepEqual(result, { b: undefined })
})
```

Actually, might have to use the older `hasOwnProperty` function for node <16 compatability.

<img width="334" alt="Screen Shot 2022-10-19 at 12 04 45 PM" src="https://user-images.githubusercontent.com/15269623/196781463-20f94c0c-8800-4b31-b049-b444826a2511.png">

Reply:
You are right! My intent was to keep undefined in the object, but my brain made a mistake... 🤦🏼

Reply:
`Object.hasOwn` requires tslib to be bumped to `es2022`, which I think is not compatible with node v12
So, I opted for the `.hasOwnProperty` method of the object class